### ISSUE TITLE
Timeout issue

### ISSUE THREAD

**Issue by Gokalex** [ToshY/BunnyNet-PHP#46]
Hello, i am getting this error during the upload:

PHP Fatal error:  Uncaught Symfony\Component\HttpClient\Exception\TransportException: Idle timeout reached for "https://video.bunnycdn.com/library/.........

I am looking for a way to set the timeout but i can't find it

**Comment by ToshY** [comment]
Hey @Gokalex 👋 

Thanks for the bug report.

Could you provide more details on the code you're executing and what you're trying to achieve?

From looking at the endpoint you've provided and taking the exception in consideration, I assume that you're uploading a (large) file and it timeouts at some point (correct me if I'm wrong).

In that case, I initially did not take this in consideration when I first wrote that part, as I did not upload large files through the API myself. So it just creates a `HttpClient` now without any specific options or the ability to pass options yourself.

I want to support PSR-18 so you can specify your own client (and options), but I haven't come around to doing so. Here's the related [issue](https://github.com/ToshY/BunnyNet-PHP/issues/27).

---

TLDR

I think that in the meantime you could set the `default_socket_timeout` to a desired value, or `-1` for infinite timeout (references: [Symfony docs](https://symfony.com/doc/current/http_client.html#dealing-with-network-timeouts); [PHP](https://www.php.net/manual/en/filesystem.configuration.php#ini.default-socket-timeout)).

**Comment by ToshY** [comment]
This will be closed when the new major release gets published, as it will be possible to supply your own client and options.
